runsc/cmd/checkpoint.go

```diff
@@ -58,6 +59,7 @@ func (*Checkpoint) Usage() string {
 func (c *Checkpoint) SetFlags(f *flag.FlagSet) {
 	f.StringVar(&c.imagePath, "image-path", "", "directory path to saved container image")
 	f.BoolVar(&c.leaveRunning, "leave-running", false, "restart the container after checkpointing")
+	f.BoolVar(&c.compression, "compression", true, "compress checkpoint image on disk")
```

Comment:
Same for this, please make it an enum flag. You can look at flags like `host-fifo` as an example.

Reply:
I have added `type CheckpointCompression int` enum/flag.

Reply:
This is good on the command-line flag setting side, but I think this can go further.

The pattern I'd like to avoid is having a `bool`-type argument in function signatures. Typically those can get confusing without some kind of comment like `SomeFunction(..., true /* compress */)` to explain what the boolean means in the context of each caller. Instead, all of these can use the enum type, which makes them immediately clearer: `SomeFunction(..., statefile.CompressionNone)`.
I think declaring a `Compression` enum type in `statefile` would work, and have all other code (other than the flag type in `cmd`) use the `statefile.Compression` type instead of `bool`. The only time where `statefile.Compression` would be translated into a boolean should be when deciding to create a `Writer` or a `SimpleWriter` inside `statefile.NewWriter`. All the rest of the code can use the enum type, including in `metadata`. This also removes the need for `statefile.CompressionFromBool`; all you'd need is `statefile.CompressionFromMetadata` (which takes in a string read from `metadata[ImageCompressed]`), and perhaps a function like `func (g *CheckpointCompression) ToStatefileCompression() statefile.Compression` in `cmd/checkpoint.go`.

Reply:
Take a look pls if the latest version reflects what you were thinking.